### ISSUE TITLE
Heavy frame drop if looking in a specific direction in The Old Hunters

### ISSUE THREAD

**Issue by itsflyas** [diegolix29/shadPS4#10]
First of all I would like to thank everyone that worked on this specific fork of ShadPS4, amazing job.
In The Old Hunters DLC there is a massive frame drop (from 60ish fps to 29) when looking in a specific direction. To be more specific the frame drop occurs when looking directly straight at the first lamp of the DLC or when looking at the door from inside the Laurence boss arena.

 I've tried  many patches, textures, builds, branches and different resolutions but nothing seems to fix it. Even at lower resolutions the frame drop persists. GPU usage seems to drop when this happens. The emulator seems to be performing different actions each time the frame drop occurs, but i will attach a screenshot of the CMD just to be sure.

https://github.com/user-attachments/assets/4a5cd55f-5221-4dbc-a14f-442b85c8692a

![image](https://github.com/user-attachments/assets/68a963b4-96e9-4aa6-8ee8-6056dc04b319)

**Comment by eFMann** [comment]
I can confirm this also happens if few other spots of the DLC and the main game. 

It seems as though CPU isn’t being utilised properly in my case when looking at these directions. CPU and GPU usage would be low and the FPS would drop to 30-40

**Comment by itsflyas** [comment]
> I can confirm this also happens if few other spots of the DLC and the main game.
> 
> It seems as though CPU isn’t being utilised properly in my case when looking at these directions. CPU and GPU usage would be low and the FPS would drop to 30-40

With more testing i found out that you are right, in my case tho the CPU usage remains constant but the GPU one drops drammatically when the frame drop occurs.

![image](https://github.com/user-attachments/assets/6bbeb8b2-ad89-4236-bf2d-819626367172)

In this example i caused the frame drop multiple times and then i looked away, you can clearly see the GPU usage going down.

**Comment by diegolix29** [comment]
Hi thanks for the feedback is happening on all my builds or just on partBB? I have pushed some fixes of fps that was making slow some areas, but i have seen also this behavior elsewhere can you do a quick test on main shad to see if it happens there? 

Edit: sorry i just saw that u have tested several branches. Thanks i will see if it has been issued on main shad so we can narrow down a fix for it. 🙏🙏

**Comment by itsflyas** [comment]
Just tested on the newest 0.5.0 of main shad and it's more or less the same, there seems to be less impact on the performance tho. On the main branch the fps will drop around 40, while on this branch somewhere around 30

**Comment by elonthelad** [comment]
Looks like viewing large portions of the complex map and geometry has an excessive load on the cpu, since as mentioned before the GPU drops in usage when this happens- indicating that the cpu cannot keep up.

Looking in that direction from this part of the map would have most of the geo and mapitems in view, but im not sure how well Bloodborne utilises LOD in its map objects

**Comment by itsflyas** [comment]
Wouldn't this cause a spike in the CPU usage tho? It was mentioned that both GPU and CPU usage would drop when the frame drop occurs

**Comment by elonthelad** [comment]
> Wouldn't this cause a spike in the CPU usage tho? It was mentioned that GPU usage would drop while CPU usage would remain constant when the frame drop occurs

an increase in cpu usage when this occurs would mean more threads are being kept busy with whatever operation is causing the framerate loss. Cpu usage remains still, gpu usage decreases, frametimes get longer = program is waiting on a particular thread in the cpu to hurry up.

**Comment by Mecex** [comment]
It might be a memory allocation issue. Checking with noclip to see the static LOD, it doesn't seem to be very different from looking on the direction of the lamp. Maybe the game is preemtively loading some more of the outside to memory, and not rendering it while keeping the current render and map on memory

**Comment by itsflyas** [comment]
i don't think it is a rendering problem tho, otherwise the frame drop would be fixed / it would be less aggressive on lower resolutions. I think it's more of an allocation problem, maybe the game needs to keep a lot of enemies IA loaded because of those hunter fightning in the square.

**Comment by Mecex** [comment]
But it also happens if you look at the left, same drop, leaving the second lamp of the DLC, and that area doesn't have many AI nor map complexity.

**Comment by leonskenedy1** [comment]
disabling Hyper Threading reduces frame drops by 30%

**Comment by eFMann** [comment]
> disabling Hyper Threading reduces frame drops by 30% and increases GPU usage

I've tested this (Disabling Hyperthreading) and this doesn't seem to fix or improve the issue we are talking about. I've attached the vid. As you can see the same thing happens, depending on where you look the GPU usage will drop as well as the FPS yet the CPU still has more juice left. I've also tested different CPU profiles i have (3.6Ghz all Core; 4ghz all core; 4.5Ghz All core) and this doesn't seem to improve this issue either. 

https://github.com/user-attachments/assets/87f4101c-92cd-460d-8f7a-7f458b1bd2ff

**Comment by itsflyas** [comment]
Running some more test with Windows's Process Explorer i have noticed a particular Thread dramatically increasing in Cycles Delta when the frame drop occurs. 
Looking away:
![image](https://github.com/user-attachments/assets/4cbc442e-c6c3-4bec-a0d3-9e26490c2d0e)

During frame drop:
![image](https://github.com/user-attachments/assets/8b32029e-59ab-449f-a88f-4081dbbedebe)

Could this be relevant?

**Comment by stevebufz** [comment]
Same thing happens with me, but I have a lower system mainly xeon 1245 v5 gtx 1050ti. For me looking at a particular direction causes this issue . I thought it was because of my low ram (12gb) but that fine. For me my fps drops from 30 to 20 or sometimes 18 which is unplayable.

**Comment by stevebufz** [comment]
It can be a huge issue for low end systems compatibility because for me my game runs fine mostly at 1080p 30 fps . Also in mergos loft base while entering the room with the mice my fps drops again to 22. Also while going to yarghul chapel lamp it drops where the amygdala firing the lasers is at. And I think I am using the main build of shadps4. I will upload a video of my issues shortly after this.

**Comment by eFMann** [comment]
After updating today to the newest release for diegolix29 release build I’ve noticed a nice performance improvement when it comes to CPU bottleneck situations. 

In my previous comment I’ve attached a video to show the specific spot in the DLC where my frames dropped to 40ish.

After the update I’m getting 50ish in the exact same location. So seems like something has worked.

**Comment by itsflyas** [comment]
> After updating today to the newest release for diegolix29 release build I’ve noticed a nice performance improvement when it comes to CPU bottleneck situations.
> 
> In my previous comment I’ve attached a video to show the specific spot in the DLC where my frames dropped to 40ish.
> 
> After the update I’m getting 50ish in the exact same location. So seems like something has worked.

I heard all the devs are focusing on improving performance across all forks, even on main. Maybe we will get an update in these days that will drastically lowers RAM and VRAM usage, who knows.

**Comment by TempleOS-Enjoyer** [comment]
> After updating today to the newest release for diegolix29 release build I’ve noticed a nice performance improvement when it comes to CPU bottleneck situations.
> 
> In my previous comment I’ve attached a video to show the specific spot in the DLC where my frames dropped to 40ish.
> 
> After the update I’m getting 50ish in the exact same location. So seems like something has worked.

Interesting.. It is the opposite for me. I reverted to https://github.com/diegolix29/shadPS4/releases/tag/release-shadPS4-2024-12-29-dceb79e because with any of the newest 2 releases, I was getting weird "slowdown" and more jittery camera in Forbidden Woods, the Hunter's Dream, even the DLC, and I don't know exactly how to describe it, it kinda feels like the camera "lags behind"? It's very weird and it definitely does not happen on that version I posted up there. Not sure what's going on..

**Comment by ares1768** [comment]
Are you using controller or the keyboard and mouse? because its  a known issue with pc controls.

**Comment by stevebufz** [comment]
I am using a counterfeit ps4 controller with a wire And without ds4

**Comment by TempleOS-Enjoyer** [comment]
> Are you using controller or the keyboard and mouse? because its a known issue with pc controls.

And I'm on a legit PS4 controller, both wired and wireless doesn't make a difference. Also have a Switch controller and an aftermarket PS4 controller, same thing happens regardless.

**Comment by ares1768** [comment]
> > Are you using controller or the keyboard and mouse? because its a known issue with pc controls.
> 
> And I'm on a legit PS4 controller, both wired and wireless doesn't make a difference. Also have a Switch controller and an aftermarket PS4 controller, same thing happens regardless.

Hmm interesting, I can only show my experience with the latest FullBB release: https://imgur.com/a/qwyTN0a
My issue can be seen in the middle of the video it looks like im stopping my mouse when looking right to left but not. It lags... (locking on to enemies feels way smoother tho.) However replying to the main topic of this post I dont know if this counts but I have the same fps drop when looking in specific directions, it can be hardly seen on the fps counter. (sry) In the beggining its around 90fps, but when looking something major it drops to 40-50ish. Could this be resolved with capping my fps back? And this happens basically in every area for me, with my setup: 6650XT R7 2700X 16Gb Ram. Settings in shadps4: Fullscreen, 900p, 60fps, vblank 2, disabled: vsync,chr.abberation,motion blur. Only must have mods: ram spike, vertex fix, sfxr.
